www/pages/plan-trip/plan-trip-controller.js

```diff
@@ -78,32 +72,31 @@ angular.module('pvta.controllers').controller('PlanTripController', function ($s
 
   //Called when this page is opened, and either a loaded trip has been queued
   //or there are no current existing parameters. Also called as a result of the
-  //newTrip method. Constructs the map, and then sets $scope.params as either default
+  //newTrip method. Constructs the map, and then sets $scope.request as either default
   //or loaded parameters
   var reload = function () {
     constructMap(defaultMapCenter);
     // All dates on this page are in Unix Epoch
-    currentDate = new Date();
-    $scope.params = {
-      name: 'New Trip',
-      time: {
-        datetime: currentDate,
-        type: 'departure',
-        asap: true
-      },
-      origin: {},
-      destination: {},
-      destinationOnly: true
-    };
+    $scope.noLocation = false;
     // If we loaded a trip (user came via
     // saved trip on My Buses), pull out
     // its details and display them.
     if (loadedTrip !== null) {
-      $scope.loaded = true;
-      $scope.params = loadedTrip;
-      $scope.params.time = loadedTrip.time;
+      $scope.request = loadedTrip;
+      $scope.request.time.option = $scope.timeOptions[$scope.request.time.option.id];
       loadedTrip = null;
-      if ($scope.params.destinationOnly) {
+
+      //Fix the time of the saved trip. If the datetime is in the future, keep it
+      //If the datetime is in the past, keep the time and update the date to either
+      //today or tomorrow
+      //If the request is 'Departing Now', do not update time
+      if ($scope.request.time.datetime < Date.now()) {
+        $scope.request.time.datetime.setDate(new Date().getDate());
+        $scope.request.time.datetime.setMonth(new Date().getMonth());
+      }
+
+      if ($scope.request.destinationOnly) {
```

Comment:
Does this mean that we have the user's current location already or that the saved trip used their current location?  Might be worth specifying in a comment.